sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2589,14 +2588,6 @@ private SqlQuerySpec createReadManyQuerySpecPartitionKeyIdSame(
             String idValue = itemIdentity.getId();
             String idParamName = "@param" + i;
 
-            PartitionKey pkValueAsPartitionKey = itemIdentity.getPartitionKey();
-            Object pkValue = ModelBridgeInternal.getPartitionKeyObject(pkValueAsPartitionKey);
-
-            if (!Objects.equals(idValue, pkValue)) {
-                // this is sanity check to ensure id and pk are the same
-                continue;
```

Comment:
the partitionKey created from spark side does not populate objectValue.
In order to let readmany work properly, either we change from spark side, or remove the checking here. 

but instead of SDK silently suppress this issue, it might be better to let backend throw exception if the data input from customer is invalid? Open for suggestions

Reply:
Agreed